### ISSUE TITLE
Implement multi-matmul

### ISSUE THREAD

**Issue by rsokl** [rsokl/MyGrad#81]
Similar to [numpy.linalg.multi_dot](https://docs.scipy.org/doc/numpy/reference/generated/numpy.linalg.multi_dot.html), we need an implementation for multi-matmul (mygrad only implements `matmul` and not `dot` for the time being, but this has no real bearing on this function). The point of this function is to order a chain of matrix multiplications to minimize the total cost of the computation. This pays off doubly when we do backprop.

The [source](https://github.com/numpy/numpy/blob/v1.15.0/numpy/linalg/linalg.py#L2433-L2541) reveals a simple algorithm under the hood for determining the optimal chain of matrix. They simply call `numpy.dot` in that order. For us, once the appropriate multiplication order is found, we can simply call `mygrad.matmul` in that order and not worry about implementing the back-prop 😁 .  

We need to take some care with [making the tensors be at least 2D](https://github.com/numpy/numpy/blob/ccfbcc1cd9a4035a467f2e982a565ab27de25b6b/numpy/linalg/linalg.py#L2522).

We should be able to test our result against:

```python
import functools
import mygrad as mg
def multi_matmul_slow(arrays): return functools.reduce(mg.matmul, arrays)
```

**Comment by IanCoolidge0** [comment]
@AFederici are you still working on this? if not I can try throwing something together over the next few days with some free time I now have

**Comment by AFederici** [comment]
@IanCoolidge0 I actually forgot about this thanks for reminding me lol but I'm going to try and get it done by Tuesday, and if not, feel free to take it.

**Comment by petarmhg** [comment]
@AFederici just want to check in, have you had the time to work on this?

**Comment by IanCoolidge0** [comment]
I'm currently knee deep in Battlecode so you guys can have a 2-way standoff instead :)

**Comment by rsokl** [comment]
Closed with #95
